modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpResponseCreator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageEncoder;
+import io.netty.handler.codec.http.DefaultHttpContent;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.DefaultLastHttpContent;
+import io.netty.handler.codec.http.HttpResponse;
+
+import java.util.List;
+
+@ChannelHandler.Sharable
+class Netty4HttpResponseCreator extends MessageToMessageEncoder<Netty4HttpResponse> {
+
+    public static final int ONE_TWENTY_EIGHT_KB = 128 * 1024;
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, Netty4HttpResponse msg, List<Object> out) {
+        HttpResponse response = new DefaultHttpResponse(msg.protocolVersion(), msg.status(), msg.headers());
+        out.add(response);
+        ByteBuf content = msg.content();
+        while (content.readableBytes() > ONE_TWENTY_EIGHT_KB) {
+            out.add(new DefaultHttpContent(content.readRetainedSlice(ONE_TWENTY_EIGHT_KB)));
```

Comment:
I wonder if can do better here. If we just slice to 128k here, we'll still see massive amounts of 128k `byte[]` allocations for large responses. Couldn't we just copy to new `ByteBuf` allocated by Netty's allocator and pre-sized to `128k` here and assume those will mostly be backed by a single `byte[]` (not sure if we can assume that I must admit but maybe there's some things we could do to make it so at least) and thus not incur the allocation in the compressor because ? 
Or asked the other way around, maybe we could just use a chunk size here that we know the Netty allocator would use single `byte[]` backed buffers for and use that?

Reply:
I'm not sure what this comment means. But I am planning on making the sliced size variable based on the underlying chunk sizes in the next iteration.

Reply:
> I'm not sure what this comment means.

sorry about that :) but:

>  But I am planning on making the sliced size variable based on the underlying chunk sizes in the next iteration.

Thanks that's what I was looking for :)